src/main/java/com/chikli/hudson/plugin/naginator/NaginatorPublisherScheduleAction.java

```diff
@@ -265,16 +267,16 @@ public CharSequence subSequence(int start, int end) {
         }
     }
     
-    private boolean parseLogImpl(final File logFile, @Nonnull final String regexp) throws IOException {
-        // TODO annotate `logFile` with `@Nonnull`
+    private boolean parseLogImpl(File logFile, Charset charset, @Nonnull final String regexp) throws IOException {
+        // TODO annotate `logFile` and 'charset' with `@Nonnull`
         // after upgrading the target Jenkins to 1.568 or later.
 
         // Assume default encoding and text files
         String line;
         Pattern pattern = Pattern.compile(regexp);
         BufferedReader reader = null;
         try {
-            reader = new BufferedReader(new FileReader(logFile));
+            reader = new BufferedReader(new InputStreamReader(new FileInputStream(logFile), charset));

```

Comment:
<BugInstance type='DM_DEFAULT_ENCODING' priority='High' category='I18N' message='Found reliance on default encoding in com.chikli.hudson.plugin.naginator.NaginatorPublisherScheduleAction.parseLogImpl(File, String): new java.io.FileReader(File)' lineNumber='277'/>